oauth2/src/main/java/com/linecorp/armeria/client/auth/oauth2/OAuth2Client.java

```diff
@@ -58,14 +60,12 @@
 
     @Override
     public HttpResponse execute(ClientRequestContext ctx, HttpRequest req) throws Exception {
-        return HttpResponse.from(authorizationGrant.withAuthorization(req).thenApply(reqWithAuth -> {
-            // Update the ctx.request
-            ctx.updateRequest(reqWithAuth);
-            try {
-                return unwrap().execute(ctx, reqWithAuth);
-            } catch (Exception e) {
-                return Exceptions.throwUnsafely(Exceptions.peel(e));
-            }
-        }));
+        final CompletionStage<HttpResponse> future = authorizationGrant.getAccessToken().thenApply(token -> {
+            final HttpRequest newReq = req.withHeaders(req.headers().toBuilder().set(
+                    HttpHeaderNames.AUTHORIZATION, token.authorization()).build());
```

Comment:
In the perspective view of code aesthetics, this does not look good. 
How about introducing a builder API that builds authorization header fluently.
For example:
```
HttpHeaders.builder()
           .authorization(token.authorization())
           .build();

```

Since it is not related to this PR, you can also handle it in a separate PR.

Reply:
Added a shortcut method for authorization header!